compose/foundation/foundation/src/uikitMain/kotlin/androidx/compose/foundation/text/input/internal/selection/TextFieldSelectionState.uikit.kt

```diff
@@ -0,0 +1,356 @@
+                    }
+
+                    2 -> {
+                        doRepeatingTapSelection(
+                            downEvent.changes.first(),
+                            selectionState,
+                            SelectionAdjustment.Word
+                        )
+                    }
+
+                    else -> {
+                        val downChange = downEvent.changes.first()
+                        clearSelection(
+                            downChange,
+                            selectionState
+                        ) // Previous selection must be cleared, otherwise this closure won't get third (and further) click
+                        doRepeatingTapSelection(
+                            downChange,
+                            selectionState,
+                            SelectionAdjustment.Paragraph
+                        )
+                    }
+                }
+            }
+        }
+    }
+}
+
+private fun doRepeatingTapSelection(
+    touchChange: PointerInputChange,
+    selectionState: TextFieldSelectionState,
+    selectionAdjustment: SelectionAdjustment
+) {
+    val selectionOffset = selectionState.textLayoutState.getOffsetForPosition(
+        position = touchChange.position
+    )
+    touchChange.consume()
+
+    val newSelection = selectionState.updateSelection(
+        selectionState.textFieldState.visualText,
+        selectionOffset,
+        selectionOffset,
+        isStartHandle = false,
+        adjustment = selectionAdjustment,
+    )
+
+    selectionState.textFieldState.selectCharsIn(newSelection)
+    selectionState.updateTextToolbarState(Selection)
+}
+
+private fun clearSelection(
+    touchChange: PointerInputChange,
+    selectionState: TextFieldSelectionState
+) {
+    val selectionOffset = selectionState.textLayoutState.getOffsetForPosition(
+        position = touchChange.position
+    )
+    val clearedSelection = selectionState.updateSelection(
+        TextFieldCharSequence(selectionState.textFieldState.visualText, TextRange.Zero),
+        selectionOffset,
+        selectionOffset,
+        isStartHandle = false,
+        adjustment = SelectionAdjustment.None,
+    )
+    selectionState.textFieldState.selectCharsIn(clearedSelection)
+}
+
+private class UIKitTextFieldTextDragObserver(
+    private val textFieldSelectionState: TextFieldSelectionState,
+    private val requestFocus: () -> Unit = {}
+) : TextDragObserver {
+    private var dragBeginPosition: Offset = Offset.Unspecified
+    private var dragTotalDistance: Offset = Offset.Zero
+
+    private fun onDragStop() {
+        // Only execute clear-up if drag was actually ongoing.
+        if (dragBeginPosition.isSpecified) {
+            textFieldSelectionState.clearHandleDragging()
+            dragBeginPosition = Offset.Unspecified
+            dragTotalDistance = Offset.Zero
+            textFieldSelectionState.directDragGestureInitiator = InputType.None
+            requestFocus()
```

Comment:
```suggestion
            requestFocus()
            textFieldSelectionState.clearHandleDragging()
```

Required for magnifier to work

Reply:
Thanks! Added